proposals/csharp-7.0/tuples.md

```diff
@@ -0,0 +1,437 @@
+
+On top of the member-wise conversions implied by target typing, implicit conversions between tuple types themselves are allowed.
+
+### Target typing
+
+> This section should be added after [Anonymous function conversions and method group conversions](../../spec.md#Anonymous-function-conversions-and-method-group-conversions)
+
+A tuple literal is "target typed" when used in a context specifying a tuple type. The tuple literal has a "conversion from expression" to any tuple type, as long as the element expressions of the tuple literal have an implicit conversion to the element types of the tuple type.
+
+```csharp
+(string name, byte age) t = (null, 5); // Ok: the expressions null and 5 convert to string and byte
+```
+
+A successful conversion from tuple expression to tuple type is classified as an *ImplicitTuple* conversion, unless the tuple's natural type matches the target type exactly, in such case it is an *Identity* conversion.
+
+```csharp
+void M1((int x, int y) arg){...};
+void M1((object x, object y) arg){...};
+
+M1((1, 2));            // first overload is used. Identity conversion is better than implicit conversion.
+M1(("hi", "hello"));   // second overload is used. Implicit tuple conversion is better than no conversion.
+```
+
+Target typing will "see through" nullable target types. A successful conversion from tuple expression to a nullable tuple type is classified as *ImplicitNullable* conversion.
+
+```csharp
+((int x, int y, int z)?, int t)? SpaceTime()
+{
+    return ((1,2,3), 7);  // valid, implicit nullable conversion
+}
+```
+
+## Additions to [Expressions](../../spec/expressions.md)
+
+### Overload resolution and tuples with no natural types
+
+> The following text is added after after the bullet list in [Exactly matching expressions](../../spec/expressions.md#Exactly-matching-expressions):
+
+The exact match rule for tuple expressions is based on the natural types of the constituent tuple arguments. The rule is mutually recursive with respect to other containing or contained expressions not in a possession of a natural type.
+
+### Deconstruction expressions
+
+> This section should be added at the end of the [Expressions](../../spec/expressions.md) chapter.
+
+Tuple deconstruction expressions receive and "split out" a tuple:
+
+```csharp
+(var sum, var count) = Tally(myValues); // deconstruct result
+Console.WriteLine($"Sum: {sum}, count: {count}");  
+```
+
+The `_` wildcard indicates that the one or more of the tuple fields are discarded:
+
+```csharp
+(var sum, _) = Tally(myValues); // deconstruct result
+Console.WriteLine($"Sum: {sum}, count was ignored");  
+```
+
+Any object may be deconstructed by providing an accessible `Deconstruct` method, either as a member or as an extension method. A `Deconstruct` method converts an object to a set of discrete values. The Deconstruct method "returns" the component values by use of individual `out` parameters. Deconstruct is overloadable.
```

Comment:
>a [](start = 91, length = 1)

an instance

Reply:
instance added in #3146 

Other comments logged in #3185